### ISSUE TITLE
java.lang.UnsatisfiedLinkError

### ISSUE THREAD

**Issue by makssof** [kristian/system-hook#13]
Hello.
Windows 10.14393 x86

```java
import lc.kra.system.mouse.GlobalMouseHook;

public class Bootstrap{
	public static void main(String[] args){
		GlobalMouseHook hook = new GlobalMouseHook();
	}
}
```

```
Exception in thread "Global Mouse Hook Thread" java.lang.UnsatisfiedLinkError: lc.kra.system.mouse.GlobalMouseHook$NativeMouseHook.registerHook()I
	at lc.kra.system.mouse.GlobalMouseHook$NativeMouseHook.registerHook(Native Method)
	at lc.kra.system.mouse.GlobalMouseHook$NativeMouseHook.run(GlobalMouseHook.java:196)
```

**Comment by kristian** [comment]
Are you trying to run the framework from source, or did you download any of the JAR bundles provided in the [releases](https://github.com/kristian/system-hook/releases) section? If you are running it from source, you'll have to download the DLL files from AppVeyor or compile them yourself, and put them to the mouse/keyboard-hook.lib.path directory. Hope this helps.

**Comment by makssof** [comment]
I tried both methods (.jar & sources)
Same error (with [keyboard || mouse])

**Comment by kristian** [comment]
How did you add the JAR file to your project, are you using Maven / Eclipse / command line? Can you please try to run the debug JARs from [AppVeyor](https://ci.appveyor.com/project/kristian/system-hook) for your platform and see if there is any more debug output in console?

**Comment by makssof** [comment]
![sh](https://cloud.githubusercontent.com/assets/6056515/23665462/837c8fac-0369-11e7-960a-ebd65dce4715.png)

**Comment by kristian** [comment]
Hmm honestly, at the moment I have no clue why this is happening. The strange thing is: the DLL seems to be loaded just fine (this is the first NATIVE print), but afterwards the native method can't be called in the DLL. But the native method name doesn't vary between platforms, so it should work in any case. I'm not able to reproduce the problem on my (x64) machine.

Could you please download the attached [system-hook-examples.zip](https://github.com/kristian/system-hook/files/826076/system-hook-examples.zip), unzip it, put the .jar file next to the system-hook-2.5.jar of your platform, run the following commands in the folder containing both files on your command line (WIN+R, CMD) and post the console output? Thanks.

```
java -version
java -cp system-hook-2.5.jar;system-hook-examples.jar lc.kra.system.keyboard.example.GlobalKeyboardExample
```

**Comment by makssof** [comment]
```
c:\test>dir /D
 Том в устройстве C имеет метку Windows
 Серийный номер тома: B225-E625

 Содержимое папки c:\test

[.]                        system-hook-2.5.jar
[..]                       system-hook-examples.jar
               2 файлов        625 198 байт
               2 папок   2 960 048 128 байт свободно

c:\test>java -version
java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) Client VM (build 25.121-b13, mixed mode)

c:\test>java -cp system-hook-2.5.jar;system-hook-examples.jar lc.kra.system.keyboard.example.GlobalKeyboardExample
NATIVE: DllMain - DLL_PROCESS_ATTACH
Exception in thread "Global Keyboard Hook Thread" java.lang.UnsatisfiedLinkError: lc.kra.system.keyboard.GlobalKeyboardHook$NativeKeyboardHook.registerHook()I
        at lc.kra.system.keyboard.GlobalKeyboardHook$NativeKeyboardHook.registerHook(Native Method)
        at lc.kra.system.keyboard.GlobalKeyboardHook$NativeKeyboardHook.run(GlobalKeyboardHook.java:174)
Global keyboard hook successfully started, press [escape] key to shutdown.

```

**Comment by kristian** [comment]
I was finally able to reproduce this issue on a 32-bit Windows 8.1 VM. I'll be looking into this now.

**Comment by kristian** [comment]
A missing linker flag in the latest MinGW environment of AppVeyor caused GCC to decorate all 32-bit functions with an `@8` suffix. This caused the native call to fail from Java because the equivalent function was missing. Fixed with 2.6 by adding `-add-stdcall-alias` linker flag.

Sorry for the late fix, but it took some time to get a 32-bit test environment going. Thanks.
